pliers/filters/text.py

```diff
@@ -60,14 +63,40 @@ def __init__(self, stemmer='porter', tokenize=True, *args, **kwargs):
                              "instance of class StemmerI.")
         self.stemmer = stemmer
         self.tokenize = tokenize
+        self.case_sensitive = case_sensitive
         super(WordStemmingFilter, self).__init__()
 
     def _filter(self, stim):
+    
+        pos_map = {
+            'ADJ':'a',
+            'ADJ_SAT': 's',
+            'ADV': 'r',
+            'NOUN': 'n',
+            'VERB': 'v'
+        }
+    
+        def pos_wordnet(txt):
+            pos_tagged = dict(nltk.pos_tag(txt, tagset='universal'))
+            pos_tagged = {tok: pos_map[tag] if tag in pos_map else 'n' for tok,tag in pos_tagged.items()}
+            return pos_tagged
+        
         if self.tokenize:
-            tokens = stim.text.split()
-            stemmed = ' '.join([self.stemmer.stem(tok) for tok in tokens])
+            tokens = [s.lower() for s in stim.text.split()] if not self.case_sensitive else stim.text.split()
```

Comment:
We might want to use nltk's tokenization functionality (minimally, `nltk.word_tokenize`) instead of just splitting on whitespace.

Reply:
check.